include/deal.II/fe/fe_hermite.h

```diff
@@ -0,0 +1,258 @@
+   */
+  virtual void
+  fill_fe_face_values(
+    const typename Triangulation<dim, spacedim>::cell_iterator &cell,
+    const unsigned int                                          face_no,
+    const hp::QCollection<dim - 1> &                            quadrature,
+    const Mapping<dim, spacedim> &                              mapping,
+    const typename Mapping<dim, spacedim>::InternalDataBase &mapping_internal,
+    const dealii::internal::FEValuesImplementation::MappingRelatedData<dim,
+                                                                       spacedim>
+      &                                                            mapping_data,
+    const typename FiniteElement<dim, spacedim>::InternalDataBase &fe_internal,
+    dealii::internal::FEValuesImplementation::FiniteElementRelatedData<dim,
+                                                                       spacedim>
+      &output_data) const override;
+
+
+
+protected:
+  /**
+   * Wrapper function for FE_Poly::get_data() that ensures relevant
+   * shape value data is copied to the new data object as well.
+   */
+  virtual std::unique_ptr<
+    typename FiniteElement<dim, spacedim>::InternalDataBase>
+  get_data(
+    const UpdateFlags             update_flags,
+    const Mapping<dim, spacedim> &mapping,
+    const Quadrature<dim> &       quadrature,
+    dealii::internal::FEValuesImplementation::FiniteElementRelatedData<dim,
+                                                                       spacedim>
+      &output_data) const override
+  {
+    std::unique_ptr<typename FiniteElement<dim, spacedim>::InternalDataBase>
+          data_ptr = FE_Poly<dim, spacedim>::get_data(update_flags,
+                                                  mapping,
+                                                  quadrature,
+                                                  output_data);
+    auto &data =
+      dynamic_cast<typename FE_Poly<dim, spacedim>::InternalData &>(*data_ptr);
+
+    const unsigned int n_q_points = quadrature.size();
+
+    if ((update_flags & update_values) &&
+        ((output_data.shape_values.n_rows() > 0) &&
+         (output_data.shape_values.n_cols() == n_q_points)))
+      data.shape_values = output_data.shape_values;
+
+    return data_ptr;
+  }
+
+
+
+private:
+  mutable Threads::Mutex mutex;
+
+  /**
+   * Variable storing the order of the highest derivative that the current
+   * @p FE_Hermite object can enforce continuity for. Here the order of
+   * derivative only counts in one spatial direction, so the derivative
+   * $\frac{d^{2}f}{dx \; dy}$ would be counted as a first order derivative
+   * of $f$, as an example.
+   */
+  unsigned int regularity;
+
+
+
+public:
+  /**
+   * Returns the regularity of the Hermite FE basis.
+   */
+  inline unsigned int
+  get_regularity() const
+  {
+    return this->regularity;
+  };
+};
+
+/** @} */
+
+
+
+DEAL_II_NAMESPACE_CLOSE
+
+#include <deal.II/numerics/vector_tools_hermite.h>
```

Comment:
```suggestion
```
What is the reason for having this include here?

Reply:
I think this was left over from when the projection functions were declared in fe_hermite to avoid changing any code that used those functions. I will remove this.